### ISSUE TITLE
Accessibility Insights Fastpass causes pages using the monaco editor to crash

### ISSUE THREAD

**Issue by mryanlo** [microsoft/accessibility-insights-web#4436]
Repro steps:
1. Open a page in some website/app that uses the monaco editor. For example: https://microsoft.github.io/monaco-editor/
2. Install the [Microsoft Accessibility Insights chrome extension](https://chrome.google.com/webstore/detail/accessibility-insights-fo/pbjjkligggfmakdaogkfomddhfmpjeni?hl=en-US). 
3. Run the Fast Pass check on the page from step 1.
![image](https://user-images.githubusercontent.com/55516393/123315005-154c0b00-d4e0-11eb-9dd2-f7ee3a893ad8.png)
4. Observe the fast pass takes longer to run and after a few seconds, the page crashes with an out of memory error:
![image](https://user-images.githubusercontent.com/55516393/123315101-2e54bc00-d4e0-11eb-979e-7bd69bbc5dde.png)

Expected: fast pass should be able to run on all parts of the page to test for accessibility issues even if the page has a monaco editor in it.
Actual: the out of memory error

We need to run this fast pass command on our app's pages to make sure they are accessibility compliant for our customers. We can't do so if the monaco editor is causing some to crash, and we're planning on adding the monaco editor to more parts of our app soon.

Browser: Google Chrome (Microsoft Edge works okay, but the fast pass will run very slowly)

I have originally filed [this bug to the microsoft monaco editor team](https://github.com/microsoft/monaco-editor/issues/2546) and they said it isn't likely an issue with the monaco editor, and asked me to file it to the accessibility insights extension team

**Comment by karanbirsingh** [comment]
I can reproduce this bug using AI-Web in Chrome. I can also reproduce this bug by running axe-core in the console:

```js
await fetch("https://cdnjs.cloudflare.com/ajax/libs/axe-core/4.2.1/axe.min.js").then(r => r.text()).then(src => eval(src));
const results = await axe.run();
console.log(results);
```

However, versions of axe-core >= 4.2.3 can scan the page using the above snippet. I did not try the axe-core update locally in the extension. I also didn't see anything notable in the [changelog](https://github.com/dequelabs/axe-core/blob/develop/CHANGELOG.md#423-2021-06-22), but it's possible this issue will be fixed after our next axe-core update.

Today, the scan works in Edge - this is probably the best workaround for now.

The scan also works using the `accessibility-insights-scan` [npm package](https://www.npmjs.com/package/accessibility-insights-scan): 
`ai-scan --url https://microsoft.github.io/monaco-editor/ --crawl --restart`

This is because we [increased the memory limit](https://github.com/microsoft/accessibility-insights-service/pull/1318) in node environments using a v8 flag. I wasn't able to use this flag against Chrome.exe outside node.

**Comment by ferBonnin** [comment]
marking this as blocked until we update to axe-core 4.3 and linking to the feature

**Comment by ferBonnin** [comment]
axe-core has been updated to 4.3.2 and I can't repro this issue anymore, marking as resolved! 
@mryanlo please let us know if this has addressed the issue 😃

**Comment by mryanlo** [comment]
@ferBonnin I have tested this and was able to successfully run the AI fastpass on pages with the monaco editor. thanks for resolving this!

**Comment by dbjorge** [comment]
Debugged an issue with a similar symptom today, but it turned out to be a different/new root cause. New issue is tracked as https://github.com/dequelabs/axe-core/issues/3923
